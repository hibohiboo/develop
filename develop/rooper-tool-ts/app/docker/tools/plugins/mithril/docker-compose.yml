version: '3'
services:

  # 構文チェック(ts)
  tslint:
    build: ./webpack
    volumes: 
      - ./config/tsconfig.json:/app/tsconfig.json
      - ./config/tslint.json:/app/tslint.json
      - ../../../../src/browser/plugins/ScenarioMaker/mithril:/app/src/ScenarioMaker
      - ../../../../src/browser/plugins/ScenarioMaker/common:/app/src/common
  # ビルドツール
  webpack:
    build: ./webpack
    volumes:
      - ./config/.babelrc:/app/.babelrc
      - ./config/tsconfig.json:/app/tsconfig.json
      - ./config/webpack.config.babel.js:/app/webpack.config.babel.js
      - ../../../../src/browser/plugins/ScenarioMaker/mithril:/app/src/ScenarioMaker
      - ../../../../src/browser/plugins/ScenarioMaker/common:/app/src/common
      - ../../../../dist/bundle-webpack:/app/dist
      - ../../../../src/browser/plugins/ScenarioMaker/public:/app/public
      # 構文チェック用設定
      - ./config/tslint.json:/app/tslint.json
    command: [npm, run, dev-server]
    ports:
      -  8080:8080

  # ユニットテスト用 jest
  jest:
    build: ./jest
    volumes:
      - ../../../../src/plugins/ScenarioMaker/mithril:/app/src/ScenarioMaker
      - ./jest/package.json:/app/package.json
      - ./config/tsconfig.json:/app/tsconfig.json
      - ../../../../dist/coverage-jest:/app/coverage
