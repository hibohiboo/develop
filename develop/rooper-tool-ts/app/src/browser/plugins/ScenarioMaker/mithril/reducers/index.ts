import { combineReducers } from 'redux';
import tragedySetList from './tragedySetList';

export default combineReducers({
  tragedySetList,
});
