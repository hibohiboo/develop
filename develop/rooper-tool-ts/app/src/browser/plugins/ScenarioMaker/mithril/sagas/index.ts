import { call, put, takeEvery } from 'redux-saga/effects';
import { FAILED, REQUEST, SUCCESS } from '../actions/tragedySet';
import { get as getSet } from '../browser/request';
import ITragedySetListItem from '../interfaces/ITragedySetListItem';

// ワーカー Saga: TRAGEDYSET_FETCH_REQUESTED Action によって起動する
function* fetchTragedySet(action: {type: string; payload: {url: string};}) {
  try {
    const tragedySetList: ITragedySetListItem[] = yield call(getSet, action.payload.url);
    yield put({ type: SUCCESS, payload:{ tragedySetList } });
  } catch (e) {
    yield put({ type: FAILED, message: e.message });
  }
}

/*
  TRAGEDYSET_FETCH_REQUESTED Action が送出されるたびに fetchTragedySet を起動します。
  ユーザ情報の並列取得にも対応しています。
*/
function* mySaga() {
  yield takeEvery(REQUEST, fetchTragedySet);
}

export default mySaga;
