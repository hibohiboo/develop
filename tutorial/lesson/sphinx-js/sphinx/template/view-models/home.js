var _ = require('lodash');
var helper = require('jsdoc/util/templateHelper');
var util = require('../util');
var logger = require('jsdoc/util/logger');

module.exports = home;

/**
 * The Mustache template to render the first page.
 * @param  {object}   context the current context
 * @param  {Function} cb      called after the generation has been done
 */
function home(context, cb) {
  var viewModel = _.extend(
    {
      package: helper.find(context.data, {kind: 'package'})[0]
    }, util.docletChildren(context, null, util.mainDocletKinds),
    util.docletChildren(context, null, util.subDocletKinds),
    util.rstMixin,
    _.pick(context, ['readme'])
  );

  logger.debug('home namespaces', viewModel.namespaces);

  util.view('home.rst', viewModel, cb);
}
