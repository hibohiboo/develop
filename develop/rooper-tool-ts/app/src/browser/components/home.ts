import * as m from 'mithril';
import nav from './nav';

const component: m.Component<{}, {}> = {
  view(vnode) {
    return m('.page', [
      m(nav),
      m('h2', 'トップ'),
      m('p', '惨劇ルーパーオンラインツールへようこそ.'),
    ]);
  },
};

export default component;
