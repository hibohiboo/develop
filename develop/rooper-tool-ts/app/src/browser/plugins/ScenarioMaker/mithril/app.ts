
import * as m from 'mithril';
import { createStore } from 'redux';
import { TragedySet } from '../common/models/TragedySet';
import { requsetTragedySetList } from './actions/tragedySet';
import { get } from './browser/request';
// import InputForm from './components/InputForm';
import TragedySetForm from './components/InputForm/TragedySetForm';
import ITragedySetListItem from './interfaces/ITragedySetListItem';
import store from './store';
import InputForm from './components/InputForm';
import Provider from './mithril-redux';

class App implements m.Component<{}, {}> {
  view(vnode): Element{
    const state = store.getState() as {tragedySetList: TragedySet[]};
    const tragedySetList = state.tragedySetList;
    return m('div',[
      m(TragedySetForm, { tragedySetList }),
      m(InputForm)
    ]);
  }
 }


const root = document.getElementById('app');
function render() {
  m.render(root, m(Provider,{store}, [m(App)]));
}

// subscribeによって、dispatchのたびにrenderが呼ばれる。
store.subscribe(render);
store.dispatch(requsetTragedySetList({ url:'/tragedySets/tragedySetList.json' }));
