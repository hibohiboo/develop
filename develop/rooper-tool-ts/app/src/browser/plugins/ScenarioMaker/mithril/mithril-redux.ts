import * as m from 'mithril';

let store;

/**
 * 役割は2つ
 * 1. Mithrilコンポーネント内でreact-reduxのconnect()関数を使えるようにすること
 * 2. ラップしたコンポーネントにstore情報を渡すこと
 * 
 * @class Provider
 * @implements {m.Component<{}, {}>}
 */
export default class Provider implements m.Component<{}, {}> {
  private props;
  private app;
  
  constructor(vnode){
    store = vnode.attrs.store;
    this.props = {
      state: store.getState()
    };
    console.log('mithril-redux-const')
    console.log(vnode)
    this.app = vnode.children[0].children;
  }
  /**
   * 
   * 
   * @param {Vnode} vnode 
   * @returns 
   * @memberof Provider
   */
  view() {
    const app = this.app;
    return m(app, {
      props: {
        state: store.getState()
      }
    })
  }
}