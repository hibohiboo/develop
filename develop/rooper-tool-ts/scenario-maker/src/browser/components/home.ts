import * as m from 'mithril';
import nav from './nav';

const component: m.Component<{}, {}> = {
  view(vnode) {
    return m('.page', [
      m(nav),
      m('h1', '惨劇RoopeRオンラインツール'),
      m('p', '惨劇ルーパーオンラインツールへようこそ.'),
    ]);
  },
};

export default component;
