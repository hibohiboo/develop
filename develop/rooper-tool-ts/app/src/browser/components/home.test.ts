
import * as m from 'mithril';
import { tidy } from "mithril-jest";
import home from './home';

const page = {
  view: () =>
    m("div", [ m(home) ])
};

describe("Home component", () => {
  it("snapshot", () => {
    const cmp = m(page);
    const html = tidy(cmp);
    expect(html).toMatchSnapshot();
  });
  it("title should be h2 with label 'トップ'", () => {
    const cmp = m(page);
    const html = tidy(cmp);
    expect(html).toContain("トップ");
  });
});