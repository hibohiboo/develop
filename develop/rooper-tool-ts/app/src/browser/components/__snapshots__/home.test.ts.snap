// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot 1`] = `
"<div>
  <div class=\\"page\\">
    <div>
      <a href=\\"#!/\\">Home</a> <span>|</span> <a href=
      \\"#!/about\\">About</a>
    </div>
    <h2>
      トップ
    </h2>
    <p>
      惨劇ルーパーオンラインツールへようこそ.
    </p>
  </div>
</div>
"
`;
